### PR TITLE
fixed schema validator branch

### PR DESCRIPTION
**Pull request by madhavikdb**
1.0.6.1.2 schema
2.manifest generator changes
3.updated example specs

### CODE REVIEW

File: service-spec/schema-validator/src/main/resources/hprof/0.6.1.2/profile-spec.json [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+          "name"
+        ],
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "name of the volume"
+          },
+          "size": {
+            "type": ["number","string"],
+            "pattern": "^(\\d+\\.)?\\d+(-(\\d+\\.)?\\d+)?$|^(\\d+\\.)?\\d+(E|P|T|G|M|K)(-(\\d+\\.)?\\d+(E|P|T|G|M|K))?$",
```
Comment (tushar-tushar) [inline service-spec/schema-validator/src/main/resources/hprof/0.6.1.2/profile-spec.json]: This should only be M and G right?

File: service-spec/schema-validator/src/main/resources/hspec/0.6.1.1/service-spec.json [hunk 2]
```diff
@@ -18,7 +18,10 @@
         },
         "args": {
           "description": "args is the list of build arguments to be passed during container image build. should be given in this format <key>=<value> ",
-          "$ref": "#/definitions/io.hyscale.api.map.string"
+          "items": {
```
Comment (tushar-tushar) [inline service-spec/schema-validator/src/main/resources/hspec/0.6.1.1/service-spec.json]: map only right?

File: workflow-controller/controller-service/pom.xml [hunk 3]
```diff
@@ -67,6 +67,7 @@
         <dependency>
             <groupId>io.hyscale</groupId>
             <artifactId>schema-validator</artifactId>
+            <version>${project.version}</version>
```
Comment (tushar-tushar) [inline workflow-controller/controller-service/pom.xml]: For version we are using dependency management in parent pom, this is not required here

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/ResourceLimitsHandler.java [hunk 4]
```diff
@@ -64,11 +65,11 @@
         ValueRange cpuRange = null;
         if (StringUtils.isNotBlank(memory)) {
             logger.debug("Preparing memory limits.");
-            memoryRange = getRange(memory, rangePattern);
+            memoryRange = getRange(memory, memoryPattern);
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/ResourceLimitsHandler.java]: Are we using these patterns anymore?

File: service-spec/schema-validator/pom.xml [hunk 5]
```diff
@@ -18,7 +18,9 @@
 		</dependency>
 		<dependency>
 			<groupId>io.hyscale</groupId>
-			<artifactId>commons</artifactId>
+			<artifactId>service-spec-commons</artifactId>
+			<version>${project.version}</version>
```
Comment (tushar-tushar) [inline service-spec/schema-validator/pom.xml]: Move version to parent pom under dependency management

File: service-spec/schema-validator/src/main/resources/hprof/0.6.1.2/profile-spec.json [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            "description": "size of volume",
+            "default": "1g"
```
Comment (tushar-tushar) [inline service-spec/schema-validator/src/main/resources/hprof/0.6.1.2/profile-spec.json]: default 1G ?

File: service-spec/schema-validator/src/main/resources/hprof/0.6.1.2/profile-spec.json [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      "description": "defines all secrets"
+    },
+    "props": {
+      "description": "props is an unstructured key value map",
+      "$ref": "#/definitions/io.hyscale.api.map.string"
+    },
+    "memory": {
+      "description": "Defines the range of memory the service has to use in the container for this environment",
+      "type": ["integer","string"],
+      "pattern": "^\\d+(-\\d+)?$|^\\d+(E|P|T|G|M|K)(-\\d+(E|P|T|G|M|K))?$"
```
Comment (tushar-tushar) [inline service-spec/schema-validator/src/main/resources/hprof/0.6.1.2/profile-spec.json]: Only M and G right?

File: service-spec/schema-validator/src/main/resources/hspec/0.6.1.1/service-spec.json [hunk 8]
```diff
@@ -27,7 +30,7 @@
       "description": "Describes artifact source and destination inside container.",
       "properties": {
         "name": {
-          "description": "defines the name of the artifact. should be unique",
+          "description": "defines the name of the artifacti. should be unique",
```
Comment (SameeraGrandhi) [inline service-spec/schema-validator/src/main/resources/hspec/0.6.1.1/service-spec.json]: typo

File: service-spec/schema-validator/src/main/resources/hspec/0.6.1.1/service-spec.json [hunk 9]
```diff
@@ -83,23 +86,15 @@
       "type": "object"
     },
     "io.hyscale.api.image.Image": {
-      "description": "Describes container Image, could be from dockerfile or buildSpec",
+      "description": "Describes container Image could be from dockerfile or buildSpec",
```
Comment (SameeraGrandhi) [inline service-spec/schema-validator/src/main/resources/hspec/0.6.1.1/service-spec.json]: Describes container image and its source, whether the image build from binaries or dockerfile with buildSpec inside image.

**Pull request closed without merging**